Joshua/Parkour.cs

```diff
@@ -0,0 +1,118 @@
+﻿using UnityEngine;
+using System.Collections;
+using UnityEngine.SceneManagement;
+
+public class Parkour : MonoBehaviour
+{
+    Transform chController;
+    private bool inside = false;
+    [SerializeField]
+    private float heightFactor = 8.0f;
+    private KeyCode shift = KeyCode.LeftShift;
+    private string parkourObj;
+    [SerializeField]
+    private float speed = 14.0f;
+    [SerializeField]
+    private float slideCooldown = 1.5f;
+    [SerializeField]
+    private float slideCooldownTime = 1f;
+
+
+    void Start(){
+
+    }
+
+
+    //check for collision with gameobject with tag
+    void OnTriggerEnter(Collider Col){
+        if (Col.gameObject.tag == "Ladder")
+        {
+            parkourObj = "Ladder";
+            inside = !inside;
+        }
+
+        if (Col.gameObject.tag == "ClimbWall")
+        {
+            parkourObj = "ClimbWall";
+            inside = !inside;
+        }
+    }
+
+
+    //check for leaving collision with gameobject with tag
+    void OnTriggerExit(Collider Col){
+        if (Col.gameObject.tag == "Ladder")
+        {
+
+            inside = !inside;
+        }
+
+        if (Col.gameObject.tag == "ClimbWall")
+        {
+
+            inside = !inside;
+        }
+    }
+
+
+    void Climbing(){
+        //check for gameobject with tag and key input, if true climb upwards
+        if (inside == true && parkourObj == "Ladder" && Input.GetKey("w"))
+        {
+            this.transform.position += Vector3.up;
+        }
+
+        if (inside == true && parkourObj == "RunWall" && Input.GetKey("w") || inside == true && parkourObj == "RunWall" && Input.GetKey("D") || inside == true && parkourObj == "RunWall" && Input.GetKey("A"))
```

Comment:
deze if statement gaat is te lang volgens de regels maar voor de rest ziet het er goed uit